### PR TITLE
1406 [Transactions] Expected Value Dropdown Selection

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds a new feature for the test spec form by displaying a value list dropdown when selecting one of the attributes, it allows users to select with value to add to the expected section from the different selected spans.

## Changes

- Adds value selection dropdown

## Fixes

- #1406 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/65e7cb843d9e4e108bb3901a1ef7b297

### CODE REVIEW

File: server/http/controller.go [hunk 1]
```diff
@@ -737,32 +754,30 @@ func (c *controller) buildDataStores(ctx context.Context, info openapi.ResolveRe
...
+					Span: span,
+				}}, ds...)
+			}
+
+			return dsList, nil
+		}
 	}
 
-	return ds, nil
+	return [][]expression.DataStore{ds}, nil
```
Comment (xoscar) [inline server/http/controller.go:780]: Now we are returning an array of strings, this is to cover the scenario where there is no span id in the context and we can return the different resolved values from the selected spans.

File: web/src/components/Editor/hooks/useTooltip.ts [hunk 2]
```diff
@@ -4,21 +4,24 @@ import {useParseExpressionMutation} from 'redux/apis/TraceTest.api';
...
       if (isSameAsPrev) return prevExpression;
 
       const parsedExpression = await parseExpressionMutation(props).unwrap();
 
       setPrevExpression(parsedExpression);
+      setPrevContext(props.context || {});
       setPrevRawExpression(props.expression || '');
     },
-    [parseExpressionMutation, prevExpression, prevRawExpression]
+    [context, parseExpressionMutation, prevContext, prevExpression, prevRawExpression]
```
Comment (xoscar) [inline web/src/components/Editor/hooks/useTooltip.ts:24]: Reruns if prev context changes

File: web/src/components/TestSpecForm/AssertionCheck.tsx [hunk 3]
```diff
@@ -109,15 +108,15 @@ export const AssertionCheck = ({
...
-        <Form.Item
-          {...field}
-          name={[name, 'right']}
-          rules={[{required: true, message: 'Expected value is required'}]}
-          data-cy="assertion-check-value"
-          style={{margin: 0}}
-        >
-          <Editor type={SupportedEditors.Expression} placeholder="Expected Value" context={editorContext} />
-        </Form.Item>
+        <AssertionCheckValue
```
Comment (xoscar) [inline web/src/components/TestSpecForm/AssertionCheck.tsx:111]: Moved this to a diff file

File: web/src/components/TestSpecForm/TestSpecForm.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ import OperatorService from 'services/Operator.service';
 import {TStructuredAssertion} from 'types/Assertion.types';
 import {singularOrPlural} from 'utils/Common';
 import AssertionCheckList from './AssertionCheckList';
-import useAssertionFormValues from './hooks/useAssertionFormValues';
+// import useAssertionFormValues from './hooks/useAssertionFormValues';
```
Comment (jorgeepc) [inline web/src/components/TestSpecForm/TestSpecForm.tsx:14]: 👀

**Review by jorgeepc — APPROVED** [review]
Looks good @xoscar great job!

**Pull request merged**